### ISSUE TITLE
Sidekiq Slow to pickup job when separate each worker for each queue

### ISSUE THREAD

**Issue by chaintng** [mperham/sidekiq#3790]
Ruby version: 4
Sidekiq Pro version(s): 3.5.1

Initializer:
1. `bundle exec sidekiq -q worker-1`
`bundle exec sidekiq -q worker-2`

2. `bundle exec sidekiq -q worker-1 -q worker-2`

I tried separate each worker for each queue in sidekiq job (1)

But i found that there is idle time between 2nd worker to pick up job, like 200 ms

Please check screenshot below

If I initialize sidekiq to catch both queue in 1 worker (2)
The idle time between 2 workers lower to 2ms.

Is it normal behavior?

![image](https://user-images.githubusercontent.com/2979072/37552298-bda38e7c-29e4-11e8-8c0c-d61fa050351b.png)

**Comment by mperham** [comment]
there’s not enough data to diagnose.

**Comment by chaintng** [comment]
@mperham could you tell me what information you may need?

**Comment by mperham** [comment]
All the info that is requested when creating a new issue.

**Comment by chaintng** [comment]
@mperham i just updated the info. please advise

**Comment by mperham** [comment]
If you have them, show me your initializer, config/initializers/sidekiq.rb or config file, config/sidekiq.yml.

**Comment by chaintng** [comment]
`config/initializers/sidekiq.rb`

```
if Rails.env.production?
  require 'sidekiq/logging/json'

  Sidekiq.logger.formatter = Sidekiq::Logging::Json::Logger.new
end

Sidekiq::Client.reliable_push! unless Rails.env.test?

redis_conn = proc {
  Redis::Namespace.new(Rails.application.config.namespace, redis: Redis.new(url: Rails.application.config.redis_sidekiq_url))
}

Sidekiq.configure_client do |config|
  config.redis = ConnectionPool.new(size: Rails.application.config.redis_sidekiq_conn_pool, &redis_conn)
end

Sidekiq.configure_server do |config|
  config.redis = ConnectionPool.new(size: Rails.application.config.redis_sidekiq_conn_pool, &redis_conn)

  case Rails.application.config.sidekiq_fetcher
  when 'timed'
    config.timed_fetch!
  when 'reliable'
    config.reliable_fetch!
  else
    config.super_fetch!
  end

  config.reliable_scheduler!
  config.failures_max_count = false
end

Sidekiq::Web.set :session_secret, ENV["secret_key_base"]
Sidekiq::Web.set :sessions, Rails.application.config.session_options
```

I used `timed_fetch` as a sidekiq fetcher strategy

**Comment by mperham** [comment]
Does super_fetch give you the same results?

**Comment by mperham** [comment]
what about with sidekiq’s default fetch?

**Comment by chaintng** [comment]
@mperham 
I tried with `reliable_fetch`, most of the time, it's fast, no idle time.
Also default fetch, (no config), most of the time, no idle time.

However, with these fetching strategy, we have another problem that
It has idle time within the worker instead when it has multiple job in queue
![image](https://user-images.githubusercontent.com/2979072/37563533-75d70e76-2ab5-11e8-8966-add28c985546.png)

**Comment by mperham** [comment]
I guess I can't really tell what the problem is.  I don't understand your pictures.

**Comment by chaintng** [comment]
Hi @mperham,
Okay, let me summarize theproblem without using picture then.

*Background*
I have 2 queues
- `worker-1`
- `worker-2`

I ran 2 sidekiq worker to consumer each queue
- `bundle exec sidekiq -q worker-1`, i will call this `Worker1`
- `bundle exec sidekiq -q worker-2`, i will call this `Worker2`

*Note:* `Worker1` will consumer job in queue `worker-1` and produce job to queue `worker-2`

But there is some idle time that i would like to avoid

*Scenario 1: * with `timed_fetch`, `super_fetch`
There is idle time between 2 worker. (like 200ms)

with `reliable_fetch` and default fetch
There is idle time between first job and second job. (like 400ms)

Could you please advise?

**Comment by mperham** [comment]
timed_fetch and reliable_fetch are no longer supported, don't use them.

I don't make any guarantees about the timing.  Sidekiq is asynchronous; Redis is single-threaded and other operations might block it for ms, Ruby GC can introduce pauses, etc.  I don't know where your latency is coming from.

**Comment by chaintng** [comment]
Ic. The system i'm taking care of right now are using sidekiq intensively.
Each queue is depending on the precedent queue
And it's critical path to customer interaction, so it's time-sensitive.

So, do you think are we abusing sidekiq right now?

**Comment by mperham** [comment]
I don't think you are doing anything wrong; I want to do everything I can to reduce latency but I can't tell if this is a Sidekiq issue or a problem in your setup.  I will try to reproduce the problem on my laptop Tuesday, I am busy tomorrow.

**Comment by mperham** [comment]
I think I found your issue; looks like super_fetch can introduce a superfluous `sleep(1)` when it does not get a job.  I'm refactoring the code now to remove this sleep.

**Comment by mperham** [comment]
Ok, issue is fixed and will be in Pro 4.0.2.

**Comment by agushermansyah621** [comment]
Bantu saya untuk gimana cara ya untuk mendapetkan fail lineApi/Api
